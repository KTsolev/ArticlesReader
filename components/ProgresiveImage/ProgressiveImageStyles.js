import { StyleSheet } from 'react-native';


export const styles = StyleSheet.create({
    imageOverlay: { position: 'absolute', left: 0, right: 0, bottom: 0, top: 0 },
    container: { backgroundColor: '#ededed', width: 65,  height: 65,  paddingRight: 20,
      borderRadius: 50 },
  });