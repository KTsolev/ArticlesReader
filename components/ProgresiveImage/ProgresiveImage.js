import React from 'react';
import { View, Animated } from 'react-native';
import { styles } from './ProgressiveImageStyles';

class ProgressiveImage extends React.Component {
  thumbnailAnimated = new Animated.Value(0);

  imageAnimated = new Animated.Value(0);

  handleThumbnailLoad = () => {
    Animated.timing(this.thumbnailAnimated, {
      toValue: 0,
    }).start();
  }

  onImageLoad = () => {
    Animated.timing(this.imageAnimated, {
      toValue: 1, 
    }).start();
  }

  render() {
    const {
      thumbnailSource,
      source,
      style,
      ...props
    } = this.props;

    const { container, imageOverlay } = styles;

    return (
      <View style={container}>
        <Animated.Image
          {...props}
          source={thumbnailSource}
          style={[style, { opacity: this.thumbnailAnimated }]}
          onLoad={this.handleThumbnailLoad}
          blurRadius={1}
        />
        <Animated.Image
          {...props}
          source={source}
          style={[imageOverlay, { opacity: this.imageAnimated }, style]}
          onLoad={this.onImageLoad}
        />
      </View>
    );
  }
}

export default ProgressiveImage;