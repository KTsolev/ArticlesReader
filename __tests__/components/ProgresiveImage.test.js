import ProgresiveImage from '../../components/ProgresiveImage/ProgresiveImage';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Animated } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';


const createTestProps = (props) => ({
    thumbnailSource: require('../../images/dummy-image.png'),
    source: 'https://tinyurl.com/y4lvzurm',
    ...props
  });

const onload = jest.fn(); 
    let props;
    let middlewares;
    let mockStore;

    beforeEach(() => {
    props = createTestProps({});
    middlewares = [thunk];
    mockStore = configureMockStore(middlewares);
    
    props.store = mockStore({
        uiElementsState: {
            showFilters: false,
            showPopUpMenu: false
        },
        articles: {
            filters: {}
        }
    });
    });
describe('<ProgressiveImage />', () => {
    test('renders the component', () => {
        const wrapper = shallow(<ProgresiveImage  {...props} onLoad={onload} />);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });
});