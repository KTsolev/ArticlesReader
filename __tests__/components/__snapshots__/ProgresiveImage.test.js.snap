// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressiveImage /> renders the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ededed",
      "borderRadius": 50,
      "height": 65,
      "paddingRight": 20,
      "width": 65,
    }
  }
>
  <AnimatedComponent
    blurRadius={1}
    onLoad={[Function]}
    source={
      Object {
        "testUri": "../../../images/dummy-image.png",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    style={
      Array [
        undefined,
        Object {
          "opacity": 0,
        },
      ]
    }
  />
  <AnimatedComponent
    onLoad={[Function]}
    source="https://tinyurl.com/y4lvzurm"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "opacity": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
